datalad_container/tests/test_run.py

```diff
@@ -191,6 +192,50 @@ def test_custom_call_fmt(path=None, local_file=None):
     assert_in('image=../sub/righthere cmd=XXX img_dspath=../sub', out['stdout'])
 
 
+@with_tree(
+    tree={
+        "overlay1.img": "overlay1",
+        "sub": {
+            "containers": {"container.img": "image file"},
+            "overlays": {"overlay2.img": "overlay2", "overlay3.img": "overlay3"},
+        },
+    }
+)
+def test_extra_inputs(path=None):
+    ds = Dataset(path).create(force=True)
+    subds = ds.create("sub", force=True)
+    subds.containers_add(
+        "mycontainer",
+        image="containers/container.img",
+        call_fmt="echo image={img} cmd={cmd} img_dspath={img_dspath} img_dirpath={img_dirpath} > out.log",
+        extra_input=[
+            "overlay1.img",
+            "{img_dirpath}/../overlays/overlay2.img",
+            "{img_dspath}/overlays/overlay3.img",
+        ],
+    )
+    ds.save(recursive=True)  # record the entire tree of files etc
+    ds.containers_run("XXX", container_name="sub/mycontainer")
+    # print(ds.repo.call_git(["ls-files"]))
+    ok_file_has_content(
+        os.path.join(ds.repo.path, "out.log"),
+        "image=sub/containers/container.img",
+        re_=True,
+    )
+    # print(ds.repo.call_git(["log"]))
+    commit_msg = ds.repo.call_git(["show", "--format=%B"])
+    # print(f"{commit_msg = }")
+    cmd, runinfo = get_run_info(ds, commit_msg)
+    # print(f"{runinfo = }")
+    assert set(
+        [
+            "sub/containers/container.img",
+            "overlay1.img",
+            "sub/containers/../overlays/overlay2.img",
+            "sub/overlays/overlay3.img",
+        ]
+    ) == set(runinfo.get("extra_inputs", set()))
+
```

Comment:
```suggestion


```

two empty lines typically to separate out.  

also please remove debugging commented out `print` statements if no loner needed for your development . We generally don't have such ones, so I don't think worth adding them here either